src/auto-complete.component.ts

```diff
@@ -447,4 +457,26 @@ export class NguiAutoCompleteComponent implements OnInit {
         );
     }
 
+    private calculateHeightFits() {
+        // Filter items height = 50px;
+        const heighByFilter = this.filters.length > 0 ? 50 : 0;
+        const heighByResultItems = this.filteredList.length * this.itemHeight;
+
+        const elBottom = this.el.getBoundingClientRect().bottom;
+        const listHeight = heighByResultItems + heighByFilter;
+        const fitsInPage =  elBottom + listHeight < window.innerHeight;
+
+        if (!fitsInPage) {
+            this.el.style.top = 'auto';
+            this.el.style.bottom = `${this.triggerInputHeight}px`;
+            const topOverflowedSection = elBottom - listHeight + this.triggerInputHeight - this.maxHeightTopGap;
```

Comment:
@ldiaz maybe in this case topOverflowedSection**Height** would carry more meaning?

Reply:
Yes that adds more meaning. updated